tpke/benches/tpke.rs

```diff
@@ -299,12 +299,77 @@ pub fn bench_share_encrypt_decrypt(c: &mut Criterion) {
     }
 }
 
+pub fn bench_recover_share_at_point(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Recover Share at Point Benchmark");
+    // Set up test conditions
+    let rng = &mut StdRng::seed_from_u64(0);
+    let msg_size = MSG_SIZE_CASES[0];
+
+    for &shares_num in NUM_SHARES_CASES.iter() {
+        let mut setup = SetupSimple::new(shares_num, msg_size, rng);
+        let threshold = setup.shared.threshold;
+        let selected_participant = setup.contexts.pop().unwrap();
+        let x_r = selected_participant
+            .public_decryption_contexts
+            .last()
+            .unwrap()
+            .domain;
+        let mut remaining_participants = setup.contexts;
+        for p in &mut remaining_participants {
+            p.public_decryption_contexts.pop();
+        }
+        group.bench_with_input(
+            BenchmarkId::new("Recover Share at Point", shares_num),
+            &shares_num,
+            |b, _| {
+                let mut rng = rand::rngs::StdRng::seed_from_u64(0);
+                b.iter(|| {
+                    let _ = black_box(recover_share_at_point::<E>(
+                        &remaining_participants[..shares_num],
+                        threshold,
+                        &x_r,
+                        &mut rng,
+                    ));
+                });
+            },
+        );
```

Comment:
I think you don't need to use `bench_with_input` here: Notice that `shares_num` passed in as a second argument is unused: `|b, _|`

```suggestion
        group.bench_function(
            BenchmarkId::new("Recover Share at Point", shares_num),
            |b| {
                let mut rng = rand::rngs::StdRng::seed_from_u64(0);
                b.iter(|| {
                    let _ = black_box(recover_share_at_point::<E>(
                        &remaining_participants[..shares_num],
                        threshold,
                        &x_r,
                        &mut rng,
                    ));
                });
            },
        );
```